### ISSUE TITLE
Add custom and dynamic authentication headers

### ISSUE THREAD

**Issue by chaodhib** [swagger-api/swagger-ui#3416]
Hi!

I'm using Swagger UI V2 (2.2.10) (until the json editor is implemented in V3 but that's not relevant to this ticket) and I need to add custom authentication headers.

First, the user needs to input 3 different fields in the UI. From these 3 values, custom code that I need to plug in will add on each 'Try it' request 4 different headers. 2 of them are simply the 2 of the 3 fields given by the user. And 2 other headers will be dynamically calculated (one will be a signed hash of different elements in the request (body, HTTP method, ..), the other is the current epoch time-stamp).

So in conclusion, I need to add in the GUI an option to input 3 global values. And I need to insert custom code that will create static and dynamic headers. Could you please give me pointers on how I could achieve those goals?

Should I better ask this question in swagger-js instead?

Thank you!

Chaouki

EDIT:

I found [this](https://github.com/swagger-api/swagger-ui/tree/2.x#custom-header-parameters---for-basic-auth-etc). It seems to be close to what I need..

**Comment by webron** [comment]
The link you've added is pretty much what you need.

**Comment by shockey** [comment]
Closing due to inactivity.

This is simply to keep our issue tracker clean - feel free to comment if there are any further thoughts or concerns, and we'll be happy to reopen this issue.

**Comment by chaodhib** [comment]
No need. I managed to make it work! Was easier than expected! Thank you.

I needed up modifying mostly https://github.com/swagger-api/swagger-ui/blob/2.x/src/main/javascript/view/AuthView.js and I used the link mentioned above.
